docs/proposals/build-skipping.adoc

```diff
@@ -0,0 +1,69 @@
+= Proposal: Multi build repo skipping
+
+== Purpose
+
+Skip builds for which nothing has changed in their working directory.
+
+== Background
+
+For a repo with a single build, the pipeline build will only be triggered if any changes trigger a new pipeline run.
+
+However if one has multiple builds in one repo, the pipeline build runs all build tasks even for those areas where nothing has changed.
+
+A simple scenario is a repo with a backend and frontend component whose build outputs are combined by a subsequent docker build in a subsequent deploy task.
+
+There is no good reason when only the frontend code changes that the backend build is redone as well and vice versa of course.
+
+== Solution
+
+All tasks shall be enabled to support build skipping.
+
+The following is added to the images of these tasks:
+
+- `git`
+
+Task parameter
+
+- `cache-output-dir` a boolean defaulting to `'true'`.
+
+The solution builds on the caching mechanism introduced in issue #147.
+
+`ods-start` supports a new area in the cache for the purpose of caching build task outputs.
+This area is below folder `.ods-cache/build-task`.
+
+If `cache-output-dir` is `'true'` build tasks shall copy their build outputs to `.ods-cache/build-task/$(git-sha-working-dir)`, where
+
+- `git-sha-working-dir` is the git sha of the working directory which is determined as follows: `git-sha-working-dir=$(git rev-parse "HEAD:$working_dir")`  (https://stackoverflow.com/a/58600111).
+
+However if `cache-output-dir` is `'true'` and there is prior build output at that location, cached files are made available in the tasks `output-dir`. The cache and output dir will be on the same PVC so that hard linking can be utilized if needed.
+
+For traceability the task result sets a parameter `build-reused`` to 'true' (or perhaps the git sha). +
```

Comment:
Do you mean a Tekton result here?

Reply:
Yes something similar to https://github.com/opendevstack/ods-pipeline/blob/02a166608294531db6166d296f1bdb4c23be8dcd/deploy/central/tasks-chart/templates/task-ods-package-image.yaml#L79

Reply:
Ok! Maybe we can reword this sentence a little to make that clearer.